scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5217 @@
+
+        in_left = item >= a if left_inclusive else item > a
+        in_right = item <= b if right_inclusive else item < b
+        return in_left & in_right
+
+
+class _RealDomain(_SimpleDomain):
+    r""" Represents a simply-connected subset of the real line.
+
+    Completes the implementation of the `_SimpleDomain` class for simple
+    domains on the real line.
+
+    Methods
+    -------
+    define_parameters(*parameters)
+        (Inherited) Records any parameters used to define the endpoints of the
+        domain.
+    get_numerical_endpoints(parameter_values)
+        (Inherited) Gets the numerical values of the domain endpoints, which
+        may have been defined symbolically.
+    contains(item, parameter_values)
+        (Inherited) Determines whether the argument is contained within the
+        domain
+    __str__()
+        Returns a string representation of the domain, e.g. "[a, b)".
+    draw(size, rng, proportions, parameter_values)
+        Draws random values based on the domain. Proportions of values within
+        the domain, on the endpoints of the domain, outside the domain,
+        and having value NaN are specified by `proportions`.
+
+    """
+
+    def __str__(self):
+        a, b = self.endpoints
+        left_inclusive, right_inclusive = self.inclusive
+
+        left = "[" if left_inclusive else "("
+        a = self.symbols.get(a, f"{a}")
+        right = "]" if right_inclusive else ")"
+        b = self.symbols.get(b, f"{b}")
+
+        return f"{left}{a}, {b}{right}"
+
+    def draw(self, size=None, rng=None, proportions=None, parameter_values=None):
+        r""" Draw random values from the domain.
+
+        Parameters
+        ----------
+        size : tuple of ints
+            The shape of the array of valid values to be drawn.
+        rng : np.Generator
+            The Generator used for drawing random values.
+        proportions : tuple of numbers
+            A tuple of four non-negative numbers that indicate the expected
+            relative proportion of elements that:
+
+            - are strictly within the domain,
+            - are at one of the two endpoints,
+            - are strictly outside the domain, and
+            - are NaN,
+
+            respectively. Default is (1, 0, 0, 0). The number of elements in
+            each category is drawn from the multinomial distribution with
+            `np.prod(size)` as the number of trials and `proportions` as the
+            event probabilities. The values in `proportions` are automatically
+            normalized to sum to 1.
+        parameter_values : dict
+            Map between the names of parameters (that define the endpoints)
+            and numerical values (arrays).
+
+        """
+        parameter_values = parameter_values or {}
+        rng = rng or np.random.default_rng()
+        proportions = (1, 0, 0, 0) if proportions is None else proportions
+        pvals = np.abs(proportions)/np.sum(proportions)
```

Comment:
I would not allow negative values in proportions, it is not intuitive. if you want to allow it, you need to divide by `sum(abs(proportions)`

Reply:
I have no idea why this happened. Maybe an accidental autocomplete before I disabled the agressive suggestions of my IDE. Removed.